pkg/cloud/aws/actuators/machine_scope.go

```diff
@@ -131,43 +135,40 @@ func (m *MachineScope) Close() {
 		return
 	}
 
-	// Sometimes when an object gets updated the local copy is out of date with
-	// the copy stored on the server. In the case of cluster-api this will
-	// always be because the local copy will have an out-of-date resource
-	// version. This is because something else has updated the resource version
-	// on the server and thus the local copy is behind.
-	// This retry function will update the resource version if the local copy is
-	// behind and try again.
-	// This retry function will *only* update the resource version. If some
-	// other data has changed then there is a problem. Nothing else should be
-	// updating the object and this function will (correctly) fail.
-	if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		m.V(2).Info("Updating machine", "machine-resource-version", m.Machine.ResourceVersion, "node-ref", m.Machine.Status.NodeRef)
-		m.Machine.Spec.ProviderSpec.Value = ext
-		m.V(6).Info("Machine status before update", "machine-status", m.Machine.Status)
-		latest, err := m.MachineClient.Update(m.Machine)
-		if err != nil {
-			m.V(3).Info("Machine resource version is out of date")
-			// Fetch and update the latest resource version
-			newestMachine, err2 := m.MachineClient.Get(m.Machine.Name, metav1.GetOptions{})
-			if err2 != nil {
-				m.Error(err2, "failed to fetch latest Machine")
-				return err2
-			}
-			m.Machine.ResourceVersion = newestMachine.ResourceVersion
-			return err
-		}
-		m.V(5).Info("Latest machine", "machine", latest)
-		// The machine may have status (nodeRef) that the latest doesn't yet
-		// have, however some timestamps may be rolled back a bit with this copy.
-		m.Machine.Status.DeepCopyInto(&latest.Status)
-		latest.Status.ProviderStatus = status
-		_, err = m.MachineClient.UpdateStatus(latest)
-		return err
-	}); err != nil {
-		m.Error(err, "error retrying on conflict")
+	m.Machine.Spec.ProviderSpec.Value = ext
+	old, err := m.MachineClient.Get(m.Machine.Name, v1.GetOptions{})
```

Comment:
Instead of doing a `Get` here, should we be storing the old value when we instantiate the scope instead?

Reply:
could do, what's the benefit of storing it when we start? seems like that would only increase the amount of time something could update it.

Reply:
I was thinking that it would ensure that we are only attempting to patch fields that we explicitly changed during the reconciliation. This would avoid us stomping a non-conflicting change that was made by a separate process. For example if another process modified annotations, labels, etc we wouldn't clobber those changes.

Reply:
So the assertion is that cluster-api will only be managing cluster-api owned fields/labels/annotations/etc and external tools will never touch the cluster-api owned fields/labels/annotations/etc. Conversely cluster-api will never update non-cluster-api owned fields/labels/annotations/etc. If these are correct then I agree with you and this should be storing an object copy at the init of scope.

Reply:
So, if you consider things like Upgrade, where we are orchestration some type of end state programmatically or end users attempting to modify fields then we definitely have to take in to account external changes.

Reply:
I would expect the changes mentioned above to be picked up during the next subsequent reconcile and not overwritten by the patch we are doing as part of the current reconciliation

Reply:
Ok, I think I'm on board. It still feels racy, but it's better than fetching right here, I agree.